boot.c

```diff
@@ -51,15 +46,15 @@ map_physical_memory_with_megapages(uintptr_t dram_base,
 
   /* root page table */
   root_page_table[RISCV_GET_PT_INDEX(ptr, 1)] =
-    ptd_create(kernel_va_to_pa(load_l2_page_table) >> RISCV_PAGE_BITS);
+    ptd_create(ppn(kernel_va_to_pa(load_l2_page_table)));
```

Comment:
All of the calls to ptd_create ignore the return value, we should either be checking it for errors or it shouldn't be returning anything.

Reply:
ptd_create returns the page table entry, and it is directly written to the page table.

Reply:
No error code is returned